viseron/components/webserver/api/v1/tuning/labels.py

```diff
@@ -0,0 +1,118 @@
+"""Label definitions for object detectors."""
+
+import logging
+import os
+
+from ultralytics import YOLO
+
+LOGGER = logging.getLogger(__name__)
+
+CODEPROJECTAI_MODELS = {
+    "ipcam-animal": [
+        "bird",
+        "cat",
+        "dog",
+        "horse",
+        "sheep",
+        "cow",
+        "bear",
+        "deer",
+        "rabbit",
+        "raccoon",
+        "fox",
+        "skunk",
+        "squirrel",
+        "pig",
+    ],
+    "ipcam-dark": ["bicycle", "bus", "car", "cat", "dog", "motorcycle", "person"],
+    "ipcam-general": [
+        "person",
+        "vehicle",
+        # ipcam-dark :
+        "bicycle",
+        "bus",
+        "car",
+        "cat",
+        "dog",
+        "motorcycle",
+    ],
+    "ipcam-combined": [
+        "person",
+        "bicycle",
+        "car",
+        "motorcycle",
+        "bus",
+        "truck",
+        "bird",
+        "cat",
+        "dog",
+        "horse",
+        "sheep",
+        "cow",
+        "bear",
+        "deer",
+        "rabbit",
+        "raccoon",
+        "fox",
+        "skunk",
+        "squirrel",
+        "pig",
+    ],
+}
+
+
+def _load_labels_from_file(file_path: str) -> list[str]:
+    """Load labels from file."""
+    try:
+        if not os.path.exists(file_path):
+            return []
+        with open(file_path, encoding="utf-8") as f:
+            return [line.strip() for line in f if line.strip()]
+    except OSError as e:
+        LOGGER.warning(f"Failed to load labels from {file_path}: {e}")
+        return []
+
+
+def _load_yolo_labels(model_path: str) -> list[str] | None:
+    """Load labels from YOLO model."""
+    try:
+        if not os.path.exists(model_path):
+            LOGGER.warning(f"YOLO model not found: {model_path}")
+            return None
+
+        model = YOLO(model_path)
+        return list(model.names.values())
+    except (OSError, RuntimeError) as e:
+        LOGGER.warning(f"Failed to load YOLO labels from {model_path}: {e}")
+        return None
+
+
+def get_available_labels(
+    component_name: str, domain_config: dict | None = None
+) -> list[str] | None:
+    """Get available labels for an object detector component."""
+    if component_name in ("darknet", "hailo"):
+        label_path = "/detectors/models/darknet/coco.names"  # Default path
+        if domain_config and "label_path" in domain_config:
+            label_path = domain_config["label_path"]
+        return _load_labels_from_file(label_path)
```

Comment:
Dont think this is needed, the label_path in the config dict is always present so you can just grab the value in domain_config["label_path"] without checking

```suggestion
        return _load_labels_from_file(domain_config["label_path"])
```

Same goes for the other object detectors that specify a label path

Reply:
Ah, that's right, I'll make the fix right away.

Reply:
solved in https://github.com/roflcoopter/viseron/pull/1165/commits/512e0fb7c9eacfa1ef9c902f41ac94bb633b151a